packages/wonder-blocks-core/util/add-style.md

```diff
@@ -0,0 +1,130 @@
+Adds a `style` property to components. A component can be either a custom React
+Component (e.g. Link, Button) or a JSX intrinsic (e.g. "div", "span").
+This helps you to manipulate and extend styles more easily by using the `styles`
+prop as an array of different style objects.
+
+### Example: Adding default styles
+
+You can create a new styled component by using the `addStyle` function. Note
+here that you can also define default styles for this component by passing an
+initial style object to this function.
+
+```jsx
+import {StyleSheet} from "aphrodite";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {addStyle} from "@khanacademy/wonder-blocks-core";
+
+const styles = StyleSheet.create({
+    // default style for all instances of StyledInput
+    input: {
+        background: Color.white,
+        border: `1px solid ${Color.offBlack16}`,
+        borderRadius: Spacing.xxxSmall,
+        fontSize: Spacing.medium,
+        padding: Spacing.xSmall,
+    }
+});
+
+const StyledInput = addStyle("input", styles.input);
+
+<StyledInput type="text" placeholder="Lorem ipsum"/>;
```

Comment:
Haha, I guess I didn't realize that we had, essentially, replicated the `styled-components` API, that's pretty great!

Reply:
I assume this comment is tongue-in-cheek since the style-components is quite different: https://www.styled-components.com/.